src/synapse/api.py

```diff
@@ -126,18 +131,26 @@ def _do_request(
     Raises:
         NetworkError: if there was an error fetching the api_url.
         UserExistsError: if there is an attempt to register an existing user.
+        UnauthorizedError: if the token is not allowed to do the operation.
 
     Returns:
         Response from the request.
     """
     try:
         session = requests.Session()
+        # By default always retry on connect (and in server errors). Failing on connect
+        # should mean that the server has not started to process the request.
+        # Only retry on read if retry set to True, as in a non-idempotent operation
+        # this may have undesired effects.
         retries = Retry(
+            connect=3,
```

Comment:
I don't remember: it shows in the logs when try to retry, right?

Reply:
Honestly, I'm curious about how the charm will behave since many events depend on API calls. I don't know if there is some kind of charm event timeout if, for example, a request takes all the retries to be completed.

Reply:
AFAIK, juju events by default do not have a timeout, so retrying three times should not be a problem (it should not add more than a few seconds, then it raises)